Hive integration

ASChilds opened the pull request:
Submitting new PR for Hive integrated Magpie

chu11 commented on bin/magpie-hive-daemon.sh:
```diff
@@ -0,0 +1,226 @@
+        echo $scriptpid > ${HIVE_PID_DIR}/hive-server.pid
+        sleep 5
+        
+        # Check to see if the HIVE_DB_NAME database exists.  If not, create it.
```
Normally this DB create would occur in ```magpie-setup-hive``` or ```magpie-run-project-hive```, but I assume it has to be done here b/c of the additional config down below in the script?

chu11 commented on doc/README.hive:
```diff
@@ -0,0 +1,162 @@
+   These instructions will inform you how to login to the master node
+   of your allocation and how to initialize your session.  Once in
+   your session, you can do as you please.  For example, you can
+   interact with the Hive database to start (bin/beeline ).  There
```
extraneous space after beeline

chu11 commented on doc/README.hive:
```diff
@@ -0,0 +1,162 @@
+   nodes.  Usually this is on a NFS home directory.  You'll also need to
+   download a binary version of PostgreSQL
+
+   See 'Convenience Scripts' in README about
```
perhaps here you should mention hive-testbench.  You can say it's an optional download, but necessary for running several of the benchmarking tests for Hive that come with magpie.

And perhaps say to see below about patching hive-testbench.

chu11 replied:
looks good, thanks!

chu11 commented on magpie-check-inputs:
```diff
@@ -1185,6 +1188,63 @@ then
+    if [ "${HIVE_TESTBENCH_TYPE}" != "tpch" ] \
+        && [ "${HIVE_TESTBENCH_TYPE}" != "tpcds" ]
+    then
+        echo "HiVE_TESTBENCH_TYPE must be set to tpch or tpcds"
```
typo, lower case i

ASChilds replied:
Corrected

chu11 commented on magpie/job/magpie-job-hive-testbench:
```diff
@@ -0,0 +1,169 @@
+
+if [ "${HIVE_MASTER_NODE}X" != "X" ]
+then
+#    hiveserver=${HIVE_MASTER_NODE}
```
??? left over debugging?

ASChilds replied:
Yes, corrected.

chu11 commented on magpie/job/magpie-job-hive-testbench:
```diff
@@ -0,0 +1,169 @@
+if [ "${HIVE_TESTBENCH_DIR}X" != "X" ]
+then
+    testbenchdir="${HIVE_TESTBENCH_DIR}"
+
```
I don't think there's a need for this else case?

chu11 commented on magpie/run/magpie-run-project-hive:
```diff
@@ -0,0 +1,238 @@
+            ${MAGPIE_SCRIPTS_HOME}/bin/magpie-postgres-daemon.sh start ${HIVE_CONF_DIR}
+
+            # Check to verify Postgres is up and running
+            $POSTGRES_HOME/bin/pg_isready -h localhost -p $PGPORT &> /dev/null
```
localhost?  should be HIVE_MASTER_NODE?

ASChilds replied:
So this I think is something to do with either postgresql or our security here.  When using HIVE_MASTER_NODE the pg_isready fails (it prints properly, checking node:port).  Even submitting manually once everything is running if we use anything other than localhost here it fails. I'm not sure why at this point, however the only thing I can think of is that its either security related (can't have pg_isready perform a loopback?) or postgres related.

chu11 commented on magpie/run/magpie-run-project-zeppelin:
```diff
@@ -82,9 +82,9 @@ Magpie_run_start_zeppelin () {
             # My rough estimate for setup time is 30 seconds
             local sleepwait=30
-
             echo "Waiting ${sleepwait} seconds to allow Zeppelin daemons to setup"
```
accidental line remove?

chu11 commented on magpie/run/magpie-run-project-zeppelin:
```diff
@@ -82,9 +82,9 @@ Magpie_run_start_zeppelin () {
-
             echo "Waiting ${sleepwait} seconds to allow Zeppelin daemons to setup"
             sleep ${sleepwait}
+            
```
accidental line add?

ASChilds replied:
Removed

chu11 commented on magpie/setup/magpie-setup-project-hive:
```diff
@@ -0,0 +1,203 @@
+    hivemetastoredir="${hivelocalscratchdirpath}/metastore"
+fi
+
+if [ "${PGDATA}X" != "X" ]
```
tabbing here and next few if branches is off.

ASChilds replied:
Fixed, thanks.

chu11 commented on submission-scripts/script-templates/magpie-hive:
```diff
@@ -0,0 +1,81 @@
+
+# Set hive job for MAGPIE_JOB_TYPE = hive
+# Use 'testbench' to run benchmark testing for Hive via hive-testbench modules
+#
```
should also document other options, like checkhiveup, setuponly, etc.

chu11 commented on magpie/job/magpie-job-hive-testbench:
```diff
@@ -0,0 +1,169 @@
+
+export PATH=${PATH}:${HIVE_HOME}/bin:${HADOOP_HOME}/bin
+
+if [ "${HIVE_TESTBENCH_TYPE}" != "tpch" ] && [ "${HIVE_TESTBENCH_TYPE}" != "tpcds" ]
```
HIVE_TESTBENCH_TYPE and HIVE_CLI_VERSION are checked in magpie-check-inputs, so they don't have to be double checked here again.

ASChilds replied:
OK, I've removed this check

chu11 commented on submission-scripts/script-templates/magpie-hive:
```diff
@@ -0,0 +1,81 @@
+
+export HIVE_TESTBENCH_TYPE="tpch"
+
+# Select the Hive CLI version.  Options are 'hive' and 'beeline'.
```
something I never thought to ask before b/c I don't know enough about hive, but why support this configuration?  Internally, does it matter to the user what Magpie chooses?  In some scripts you hardcode beeline.  Obviously if the user wishes to interact with Hive, they can script on their own.  But not sure why the user needs to select this?

ASChilds replied:
Originally I left this in because hive-testbench supported both, and currently Hive supports both.  At some point the hive-cli is going to be depreciated however I haven't seen anything in the documentation as to when?

chu11 replied:
ok, no problem leaving it in.

chu11 commented on submission-scripts/script-templates/magpie-hive:
```diff
@@ -0,0 +1,81 @@
+export HIVE_METASTORE_DIR="LUSTREDIRPREFIX/hive/database/postgres/data"
+
+# Configurations for Postgres, used as a backend for HiveServer/Hive Metastore
+export POSTGRES_HOME="POSTGRESDIRPREFIX/base/pgsql"
```
minor nit, no need to add ```base```.  I assume "base" was just something in your environment?

chu11 commented on submission-scripts/script-templates/magpie-hive:
```diff
@@ -0,0 +1,81 @@
+#
+# Ensure the build matches the Hadoop/HDFS version this will run against.
+#
+export HIVE_HOME="HIVEDIRPREFIX/base/apache-hive-${HIVE_VERSION}-bin"
```
nit, for consistency, remove "base" in this string.

chu11 commented on submission-scripts/script-templates/magpie-hive:
```diff
@@ -0,0 +1,81 @@
+# The following are configurations for running the hive-testbench software
+#
+# Path to the hive-testbench software
+export HIVE_TESTBENCH_DIR="TESTBENCHDIR/base/hive-testbench-hdp3"
```
nit, for consistency, remove "base" in this string.

Also, this string isn't being generated properly.  I think it should be "TEST_BENCH_PREFIX"?

chu11 commented:
Tried to run and hit

```
LOG:  unrecognized configuration parameter "dynamic_shared_memory_type" in file "/p/lcratery/achu/testing/hive/database/postgres/data/postgresql.conf" line 127
FATAL:  configuration file "/p/lcratery/achu/testing/hive/database/postgres/data/postgresql.conf" contains errors
```

I assume its an issue with postgres versions?

Perhaps the postgres.conf should only contain the minimum config necessary and not all the extraneous options?

I commented out that line, then the next time I got:

```
Testbench query count ( 24 ) greater than number of .sql files available.  Submitting maximum available ( 22 ) queries instead.
Running /g/g0/achu/bigdata/apache-hive-2.3.2-bin/bin/beeline -u jdbc:hive2://opal173:10000/tpch_flat_orc_4 -i /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings -f /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/tpch_query1.sql, logging to /tmp/achu/hive/test/16825/log/16825_qry1_results.txt
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/apache-hive-2.3.2-bin/lib/log4j-slf4j-impl-2.6.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/hadoop-2.7.0/share/hadoop/common/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Connecting to jdbc:hive2://opal173:10000/tpch_flat_orc_4
18/05/08 11:53:34 [main]: WARN jdbc.HiveConnection: Failed to connect to opal173:10000
Error: Could not open client transport with JDBC Uri: jdbc:hive2://opal173:10000/tpch_flat_orc_4: Failed to open new session: Database tpch_flat_orc_4 does not exist (state=08S01,code=0)
No current connection
```

??

ASChilds commented:
I've committed most of the updates, but I had a couple of questions...
- as far as permissions go, I haven't changed anything new?  What should they be set to?  Currently my magpie-hive-daemon.sh/magpie-postgres-daemon.sh are both 770? \
- Was there anything further up in your log during your test run to indicate an error?  I had encountered an issue like that previously but it was caused by the tpch-submit-beeline.sh script using the DB in the JDBC URI and also having it in the command.  After I updated it there wasn't an issue and I can't replicate it on my end.  I've updated the postgresql.conf to only have what we configure in it and performed a test run, looks OK here.  What version of postgresql are you using?

chu11 commented:
> as far as permissions go, I haven't changed anything new? What should they be set to?

Most of the files (atleast when I do a checkout) are at 755.  That way they are executable.  Are they atleast executable in your tree when you do a commit/push?

My postgres version is 9.2.23-3.el7_4.  I'm using the one provided by redhat.

> Was there anything further up in your log during your test run to indicate an error?

Ahh, I missed seeing this line.

```
/g/g0/achu/bigdata/magpie/magpie/job/magpie-job-hive-testbench: line 114: /g/g0/achu/bigdata/hive-testbench/tpch-setup-beeline.sh: Permission denied
```

and sure enough, the permissions were 644.  I will try again when my test cluster is back up.

P.S. I think you forgot to do the ```should also document other options, like checkhiveup, setuponly, etc.``` review comment from above.

ASChilds commented:
I've gone through and updated a couple items...there was an issue we ran into here during the DB update procedures that I've resolved.  

All of my files are set to 770 here to enable others to test/run the stack.  During the commit I see messages like:
...
 create mode 100644 testsuite/test-generate-hive.sh
 create mode 100644 testsuite/test-submit-hive.sh
...
from Git so I'm guessing its automatically setting them to 644 during the commit?  Might be a security thing?

chu11 commented:
Looking online, it appears in git that the only valid permissions are 644 and 755.  So when you have 770, git probably converts it.  If you do 755, things should work.  Could you try that?

chu11 commented on testsuite/test-validate.sh:
```diff
@@ -558,6 +558,19 @@ __check_exports_hbase () {
     fi
 }
 
+__check_exports_hive () {
```
Could this test check be more significant?   Like is there a text that ensures the benchmark ran?  Like "read 1000 rows" or something like that?

chu11 commented:
I have a run of hive that sort of seems to work, although it timed out before the job could complete (job was 1 hour).

1) How long does "testbench" normally take for you?

2) My runs seem to have problems when the PGDATA path already exists.  Can you run over and over again on the same path?

3) Hit

"Error: Error while processing statement: Cannot modify hive.security.authorization.enabled at runtime. It is in the list of parameters that can't be modified at runtime (state=42000,code=1)"

Not sure if this is an issue or not.  Do you see this normally?

chu11 commented on submission-scripts/script-templates/magpie-hive:
```diff
@@ -0,0 +1,92 @@
+# The following are configurations for running the hive-testbench software
+#
+# Path to the hive-testbench software
+export HIVE_TESTBENCH_DIR="TESTBENCHDIRPREFIX/hive-testbench-hdp3"
```
1 question/nit, why does this path have the "-hdp3" extension to it?  Every build I see only just ends with "hive-testbench".

I notice no download path for hive-testbench in README.hive.  Is there a specific build you use/downloaded?

chu11 commented:
Just documenting some things that were discussed offline:

the "default" settings of hive should be tested and added to the ```test-generate-default.sh``` script and ```test-submit-default.sh``` scripts.

Also, tests should be added to the ```test-*-cornercase.sh``` tests.  There are many hive equivalent tests that can be added in there.

chu11 commented on magpie/run/magpie-run-project-hive:
```diff
@@ -0,0 +1,238 @@
+        echo "*******************************************************"
+        echo "* Running CheckHiveUp"
+        echo "*******************************************************"
+        ${MAGPIE_SCRIPTS_HOME}/magpie/run/magpie-run-execute script ${MAGPIE_SCRIPTS_HOME}/magpie/job/magpie-job-hive-checkhiveup.py &
```
should not be ".py" extension?

chu11 commented:
The ```magpie.sbatch-srun-hadoop-and-hive``` submission script didn't have Zookeeper in it, which took me awhile to realize that was probably a problem.  After fixing that, I still can't get ```testbench``` to run correctly.  I regularly hit:

```
Connecting to jdbc:hive2://opal125:10000/tpch_flat_orc_1
18/05/25 02:46:04 [main]: WARN jdbc.HiveConnection: Failed to connect to opal125:10000
Error: Could not open client transport with JDBC Uri: jdbc:hive2://opal125:10000/tpch_flat_orc_1: Failed to open new session: Database tpch_flat_orc_1 does not exist (state=08S01,code=0)
No current connection
```

I'm thinking there is some setup and/or config problem that doesn't allow this to run as expected?

ASChilds commented:
I'm working on updating with your comments.  In the meantime I think I may know what's causing your issue with testbench...if you scroll further up the log, do you see 

"Attempting to execute /usr/var/tmp/atc/ecosystem/base/hive-testbench/tpch-setup-beeline.sh opal125 10000 1 on opal125
Scale factor must be greater than 1"

Prior to your error (it may be 5-10 lines above the connection error)?  If so, that's a limitation of the hive-testbench.   The data generated for the test must be greater than 1...so minimum of 2GB. This is a good thing to have found, I'm going to put a check in that verifies the data size in magpie-check-inputs now.  

Let me know if that's the case.

chu11 commented:
Unfortunately I deleted some of those prior slurm logs.  But that's a likely culprit.  I'll take a look again when you've pushed an update.  Thanks.

ASChilds commented:
I'm pretty sure I've corrected everything from above.  Also added some additional error/issue checks such as the 1GB testbench issue and if testbench fails to create a database.

chu11 commented:
Still getting:

```
-rw-r--r-- 1 achu achu  6483 Jun 13 13:19 magpie-setup-project-hive
```

do you have those files set to 755 now?

chu11 commented:
So I got a run that seems to sort of work.  The first query seems to work, but the second and third don't.  I think we're on the right path though?

```
*******************************************************
* Submitting Hive Testbench queries
*******************************************************
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/apache-hive-2.3.2-bin/lib/log4j-slf4j-impl-2.6.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/workspace/wsa/achu/bigdata/hadoop-2.7.0/share/hadoop/common/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Connecting to jdbc:hive2://opal110:10000
Connected to: Apache Hive (version 2.3.2)
Driver: Hive JDBC (version 2.3.2)
Transaction isolation: TRANSACTION_REPEATABLE_READ
4 rows selected (67.095 seconds)
Beeline version 2.3.2 by Apache Hive
Closing: 0: jdbc:hive2://opal110:10000
Check shows: | tpch_flat_orc_2  |
Running /g/g0/achu/bigdata/apache-hive-2.3.2-bin/bin/beeline -u jdbc:hive2://opal110:10000/tpch_flat_orc_2x -i /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings -f /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/tpch_query1.sql, logging to /tmp/achu/hive/test/24243/log/24243_qry1_results.txt
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/apache-hive-2.3.2-bin/lib/log4j-slf4j-impl-2.6.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/workspace/wsa/achu/bigdata/hadoop-2.7.0/share/hadoop/common/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Connecting to jdbc:hive2://opal110:10000/tpch_flat_orc_2x
18/06/13 15:58:20 [main]: WARN jdbc.HiveConnection: Failed to connect to opal110:10000
Error: Could not open client transport with JDBC Uri: jdbc:hive2://opal110:10000/tpch_flat_orc_2x: Failed to open new session: Database tpch_flat_orc_2x does not exist (state=08S01,code=0)
No current connection
Running /g/g0/achu/bigdata/apache-hive-2.3.2-bin/bin/beeline -u jdbc:hive2://opal110:10000/tpch_flat_orc_2x -i /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings -f /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/tpch_query2.sql, logging to /tmp/achu/hive/test/24243/log/24243_qry2_results.txt
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/apache-hive-2.3.2-bin/lib/log4j-slf4j-impl-2.6.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/workspace/wsa/achu/bigdata/hadoop-2.7.0/share/hadoop/common/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Connecting to jdbc:hive2://opal110:10000/tpch_flat_orc_2x
18/06/13 15:58:22 [main]: WARN jdbc.HiveConnection: Failed to connect to opal110:10000
Error: Could not open client transport with JDBC Uri: jdbc:hive2://opal110:10000/tpch_flat_orc_2x: Failed to open new session: Database tpch_flat_orc_2x does not exist (state=08S01,code=0)
No current connection
```

earlier in the slurm output I saw this:

```
lat_orc_2> 0: jdbc:hive2://opal110:10000/tpch_flat_orc_2> Data loaded into database tpch_flat_orc_2.                                                                                                               
```

is the database name simply wrong?

ASChilds commented:
I do have them set to 755 on my machine, I'm not sure why Git is defaulting everything new to 644.  

As far as the error, we were thwarted by leftover testing (I had an 'x' in the code to intentionally break it.  Thus the tpch_flat_orc_2x).  I've updated and re-pushed.  Sorry about that!

chu11 commented:
Try doing what this stackoverflow says for permissions:

https://stackoverflow.com/questions/40978921/how-to-add-chmod-permissions-to-file-in-git/40979016

chu11 commented:
So, I think I'm getting close.   I get the following, then things just hang until the job is killed due to time limits.  Not sure if hive is hanging, a testbench script, a script within magpie?

```
*******************************************************
* Submitting Hive Testbench queries
*******************************************************
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/apache-hive-2.3.2-bin/lib/log4j-slf4j-impl-2.6.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/workspace/wsa/achu/bigdata/hadoop-2.7.0/share/hadoop/common/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Connecting to jdbc:hive2://opal110:10000
Connected to: Apache Hive (version 2.3.2)
Driver: Hive JDBC (version 2.3.2)
Transaction isolation: TRANSACTION_REPEATABLE_READ
4 rows selected (67.113 seconds)
Beeline version 2.3.2 by Apache Hive
Closing: 0: jdbc:hive2://opal110:10000
Check shows: | tpch_flat_orc_2  |
Running /g/g0/achu/bigdata/apache-hive-2.3.2-bin/bin/beeline -u jdbc:hive2://opal110:10000/tpch_flat_orc_2 -i /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings -f /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/tpch_query1.sql, logging to /tmp/achu/hive/test/24275/log/24275_qry1_results.txt
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/apache-hive-2.3.2-bin/lib/log4j-slf4j-impl-2.6.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/workspace/wsa/achu/bigdata/hadoop-2.7.0/share/hadoop/common/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Connecting to jdbc:hive2://opal110:10000/tpch_flat_orc_2
Connected to: Apache Hive (version 2.3.2)
Driver: Hive JDBC (version 2.3.2)
Transaction isolation: TRANSACTION_REPEATABLE_READ
Running init script /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings
No rows affected (0.055 seconds)
No rows affected (0.003 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.003 seconds)
No rows affected (0.002 seconds)
Hive-on-MR is deprecated in Hive 2 and may not be available in the future versions. Consider using a different execution engine (i.e. spark, tez) or using Hive 1.X releases.
No rows affected (0.004 seconds)
No rows affected (0.001 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
Error: Error while processing statement: Cannot modify hive.security.authorization.enabled at runtime. It is in the list of parameters that can't be modified at runtime (state=42000,code=1)
```

ASChilds commented:
I'm relatively positive this is because of the hive.security.authorization.enabled setting in: 
'/g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings'

I believe by default this is enabled within the hive-testbench package.  Hive no longer allows security modifications during runtime (understandably, as it could be a vulnerability).  If you remark the following lines (add a -- to them) in the testbench.settings file, it should resolve this issue:

set hive.security.authorization.enabled=false;
set hive.security.metastore.authorization.manager=org.apache.hadoop.hive.ql.security.authorization.StorageBasedAuthorizationProvider;

let me know if this works!

chu11 commented:
Do you believes those configs are necessary for testbench to work normally?  If so we should document in README.hive.

I tried again, and those error messages are gone.  But the job still appears to hang.  Any ideas?  I unfortunately don't have any debug logs, I guess the hang is in a way such that the gather logs script isn't running.  Not sure why the gather logs script isn't running, that's probably another investigation for later.

```
*******************************************************
* Submitting Hive Testbench queries
*******************************************************
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/apache-hive-2.3.2-bin/lib/log4j-slf4j-impl-2.6.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/workspace/wsa/achu/bigdata/hadoop-2.7.0/share/hadoop/common/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Connecting to jdbc:hive2://opal110:10000
Connected to: Apache Hive (version 2.3.2)
Driver: Hive JDBC (version 2.3.2)
Transaction isolation: TRANSACTION_REPEATABLE_READ
4 rows selected (68.095 seconds)
Beeline version 2.3.2 by Apache Hive
Closing: 0: jdbc:hive2://opal110:10000
Check shows: | tpch_flat_orc_2  |
Running /g/g0/achu/bigdata/apache-hive-2.3.2-bin/bin/beeline -u jdbc:hive2://opal110:10000/tpch_flat_orc_2 -i /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings -f /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/tpch_query1.sql, logging to /tmp/achu/hive/test/24279/log/24279_qry1_results.txt
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/apache-hive-2.3.2-bin/lib/log4j-slf4j-impl-2.6.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/workspace/wsa/achu/bigdata/hadoop-2.7.0/share/hadoop/common/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Connecting to jdbc:hive2://opal110:10000/tpch_flat_orc_2
Connected to: Apache Hive (version 2.3.2)
Driver: Hive JDBC (version 2.3.2)
Transaction isolation: TRANSACTION_REPEATABLE_READ
Running init script /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings
No rows affected (0.049 seconds)
No rows affected (0.003 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.003 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.003 seconds)
No rows affected (0.002 seconds)
Hive-on-MR is deprecated in Hive 2 and may not be available in the future versions. Consider using a different execution engine (i.e. spark, tez) or using Hive 1.X releases.
No rows affected (0.004 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.001 seconds)
No rows affected (0.001 seconds)
No rows affected (0.001 seconds)
No rows affected (0.001 seconds)
No rows affected (0.001 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
No rows affected (0.002 seconds)
slurmstepd: error: *** JOB 24279 ON opal110 CANCELLED AT 2018-06-19T11:44:28 DUE TO TIME LIMIT ***
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
slurmstepd: error: *** STEP 24279.5 ON opal110 CANCELLED AT 2018-06-19T11:44:28 DUE TO TIME LIMIT ***
```

chu11 commented:
My job was set to run about an hour, which should be enough?

Yeah, something like that is fine.  Just didn't know I had to comment those things out in hive-testbench.

chu11 commented:
so I've been busy with other things.  Just a re-ping on my question above.

ASChilds commented:
I totally apologize, for some reason I thought I was waiting to hear back from you about something.  An hour should be plenty of time.  I ran it using an 8 node cluster setup here with HBase, Phoenix, Zeppelin, Spark, Hive, and Kafka all enabled (to maximize startup time) with a 2GB testbench setup and it easily completed in under an hour.  

Was your previous attempt a 'clean' run (no previous local/shared data around)?  I'm not sure other than resources why it would take that long once the testbench queries have actually started.

chu11 commented:
yeah, it was a clean run.  Are the "No rows affected" normal?

ASChilds commented:
Yes, that is the normal output from Hive.  Can you try raising the job to 2 hours and see if that resolves the issue?

chu11 commented:
the 2 hours didn't work.  It did suddenly occur to me that 

```
slurmstepd: error: *** JOB 24279 ON opal110 CANCELLED AT 2018-06-19T11:44:28 DUE TO TIME LIMIT ***
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
slurmstepd: error: *** STEP 24279.5 ON opal110 CANCELLED AT 2018-06-19T11:44:28 DUE TO TIME LIMIT ***
```

Magpie didn't catch this job running long and kill it.  Due to slurm killing the job, none of my testing logs are saved off.  Not sure why Magpie isn't catching the long running-ness and killing the job.  But will require further investigation.

On side note, @bpanneton said in another issue he wants to try and get your hive work into Magpie 2.0?  Would you be ok if Hive went into Magpie 2.0 instead of Magpie 1.0?  I know there's a lot of work you put in, but hopefully that work paid off for the Magpie 2.0 side.

Basically, I have a mini-desire to get Magpie 2.0 into master mini-shortly (shortly can still be 1-2 months), but that would require A) figuring out why its still failing for me B) merging this into 1.0, then C) upgrading the patch to Magpie 2.0.  But if Brian is already working on it for Magpie 2.0, that might be a better path for integration.

ASChilds commented:
I'll verify with @bpanneton but I don't think that will be an issue.  I do agree, it seems like the more logical approach to get it working in magpie2.0 and submit there than working through 1.0.
Plus one possible outcome is that Hive *may* function using Spark as the execution engine...that should increase the processing speed for the queries.
I'll keep you posted.

chu11 commented:
ok.  If I feel its time to go forward with merging the ```magpie20``` branch into master, and this PR hasn't been merged in, I don't have to feel too bad that I didn't get this merged in time :-)

chu11 commented:
I'm gearing for the ```magpie20``` branch to be merged into master shortly (like within a day or two), so closing this issue.  Will await the Hive PR for magpie 2.0.
